### PR TITLE
Create 2016-08-25-simple-sink-source-with-graphstage.md

### PR DESCRIPTION
**Pull request by ktoso**
Please final review

### CODE REVIEW

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      }
+    };
+  }
+}
+
+// running it would be as simple as:
+ActorSystem system;
+Materializer mat;
+
+val runnable = Source.fromGraph(new RandomNumberSource).take(10).to(Sink.ignore)
```
Comment (patriknw) [inline _posts/2016-08-25-simple-sink-source-with-graphstage.md]: why scala?

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+          }
+        });
+      }
+    };
+  }
+}
+
+// running it would be as simple as:
+ActorSystem system;
+Materializer mat;
```
Comment (patriknw) [inline _posts/2016-08-25-simple-sink-source-with-graphstage.md]: create the real thing?

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+// we can materialize the same stream multiple times:
+runnable.run(mat);
+runnable.run(mat);
+runnable.run(mat);
+
+// which would result in 3 streams running in parallel.
+```
+
+[https://gist.github.com/johanandren/a5b9e4f88fc9fc6ce56dac10bb81a50e](https://gist.github.com/johanandren/a5b9e4f88fc9fc6ce56dac10bb81a50e)
```
Comment (patriknw) [inline _posts/2016-08-25-simple-sink-source-with-graphstage.md]: shouldn't this link have text "complete source code" or something

**Comment by patriknw** [comment]
LGTM, after small nitpicks

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md (hunk 1 above)
Comment (ktoso) [reply _posts/2016-08-25-simple-sink-source-with-graphstage.md]: Whoops, sorry. Adding the type

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md (hunk 2 above)
Comment (ktoso) [reply _posts/2016-08-25-simple-sink-source-with-graphstage.md]: ok

File: _posts/2016-08-25-simple-sink-source-with-graphstage.md (hunk 3 above)
Comment (ktoso) [reply _posts/2016-08-25-simple-sink-source-with-graphstage.md]: good

**Comment by ktoso** [comment]
Simplified Java example, OK to publish?

**Comment by drewhk** [comment]
No

**Comment by drewhk** [comment]
Why Integer.Max_value? You can very easily exhaust that range.

**Comment by drewhk** [comment]
Roughly 100-200 seconds can be enough at peak perf. What was wrong with the Iterator and using a Long?

**Comment by ktoso** [comment]
With new Iterator? That's very non nice and showing that were missing API IMHO.

**Comment by drewhk** [comment]
The current one is something that invites patterns that might unexpectedly break. I don't see Iterator as non-nice at all. It is a very simple anonymous class.

**Comment by drewhk** [comment]
Eh, it has extra methods I see: https://docs.oracle.com/javase/8/docs/api/java/util/Iterator.html

**Comment by drewhk** [comment]
ah, but they are default methods. I still opt for the Iterator then.

**Comment by drewhk** [comment]
``` java
        Source.fromIterator(() -> new Iterator<Long>() {
          private long counter = 0;
          @Override public boolean hasNext() { return true; }
          @Override public Long next() { return counter++; }
        });
```

**Comment by ktoso** [comment]
This smells missing-simple-api to me but ok.

**Comment by drewhk** [comment]
The others don't (unfold, or range which is finite)? Also, this shows a nice way to build simple sources in one go, as Iterators are a model that people already likely know.

**Comment by drewhk** [comment]
We can update the post once we add a zipWithIndex

**Comment by ktoso** [comment]
Found what I was looking for: https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html#iterate-int-java.util.function.IntUnaryOperator-

**Comment by drewhk** [comment]
Ok, that might be the nicest.

**Comment by drewhk** [comment]
Use a LongStream though...

**Comment by ktoso** [comment]
By democratic vote the version with Iterator won the ms. universe contest:
:point_up: [August 25, 2016 4:24 PM](https://gitter.im/akka/dev?at=57beff91f066bd731b49ad0d)

**Pull request merged**